spec/integration/util/windows/adsi_spec.rb

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/util/windows'
+
+def fail_if_utf8_codepage
+  if Puppet.features.microsoft_windows? && Encoding.default_external == Encoding::UTF_8
+    raise 'This test must be run in a codepage other than 65001 to validate behavior'
```

Comment:
Not sure if this should be a `raise` e.g. Later testing on Nano server uses 65001 as the default codepage.